src/btree/bt_page.c

```diff
@@ -64,8 +64,8 @@ __wt_page_block_meta_assign(WT_SESSION_IMPL *session, WT_PAGE_BLOCK_META *meta)
  *     Create or read a page into the cache.
  */
 int
-__wt_page_alloc(
-  WT_SESSION_IMPL *session, uint8_t type, uint32_t alloc_entries, bool alloc_refs, WT_PAGE **pagep)
+__wt_page_alloc(WT_SESSION_IMPL *session, uint8_t type, uint32_t alloc_entries, bool alloc_refs,
+  WT_PAGE **pagep, uint32_t flags)
```

Comment:
Can we just pass in a bool instead of flags? We only have one flag now?

Reply:
One other reason I added flag was to be able identify which pages were built using the deltas.

Reply:
Or we merge the alloc_refs into flags?

Reply:
I haven't changes the alloc_refs logic. I have only added an if conditional to skip all of those steps if we were building a page using deltas.

Reply:
Yes, I understand. My suggestion is that if you are using a flags here. Better just make the alloc_ref bool also a flag.

Reply:
Can we consider this?

Reply:
As discussed, we wouldn't wanna retain it as a flag as we do not care about it's value throughout the lifetime of the struct.